internal/services/stock_exchange/initialise.go

```diff
@@ -0,0 +1,207 @@
+	"fmt"
+	"github.com/google/uuid"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type Response struct {
+	Results []Values `json:"results"`
+}
+type Values struct {
+	Ticker string `json:"ticker"`
+}
+type StockFeed struct {
+	Open      float64 `json:"open"`
+	PrevClose float64 `json:"close"`
+	Ticker    string  `json:"symbol"`
+	LTP       float64 `json:"afterHours"`
+	High      float64 `json:"high"`
+	Low       float64 `json:"low"`
+}
+
+func RandomizerAlgo() {
+
+	var allStocks []model.Stocks
+	config.DB.Model(&allStocks)
+	orderType := []string{"Limit", "Market"}
+	for _, stock := range allStocks {
+
+		//placing buy order
+		orderID := uuid.New().String()
+		rand.Seed(time.Now().UnixNano())
+		idx := rand.Intn(2)
+		order := orderType[idx]
+		min := stock.LTP - int(float64(stock.LTP)*0.01)
+		max := stock.LTP + int(float64(stock.LTP)*0.01)
+		buyOrderBody := OrderRequest{
+			OrderID:         orderID,
+			StockName:       stock.StockName,
+			OrderPlacedTime: time.Time{},
+			OrderType:       order,
+			LimitPrice:      rand.Intn(max-min+1) + min,
+			Quantity:        rand.Intn(100) + 1,
+		}
+		_, err := BuyOrder(buyOrderBody)
+		if err != nil {
+			fmt.Println(err.Error())
+			return
+		}
+
+		//placing sell order
+		orderID = uuid.New().String()
+		rand.Seed(time.Now().UnixNano())
+		idx = rand.Intn(2)
+		order = orderType[idx]
+		min = stock.LTP - int(float64(stock.LTP)*0.01)
+		max = stock.LTP + int(float64(stock.LTP)*0.01)
+		time.Sleep(1 * time.Second)
+		sellOrderBody := OrderRequest{
+			OrderID:         orderID,
+			StockName:       stock.StockName,
+			OrderPlacedTime: time.Time{},
+			OrderType:       order,
+			LimitPrice:      rand.Intn(max-min+1) + min,
+			Quantity:        rand.Intn(100) + 1,
+		}
+		_, err = SellOrder(sellOrderBody)
+		if err != nil {
+			fmt.Println(err.Error())
+			return
+		}
+	}
+	// sleep and run again
+	time.Sleep(120 * time.Second)
+	RandomizerAlgo()
+}
+
+func GetTickers(limit int) (tickers []string) {
+	urlTickers := "https://api.polygon.io/v3/reference/tickers?active=true&sort=primary_exchange&order=asc&limit=" + strconv.Itoa(limit) + "&apiKey=721mkXq0CBNvCMi5iyJ9E1gBRDiFcT8b"
+	req, _ := http.NewRequest("GET", urlTickers, nil)
+
+	res, _ := http.DefaultClient.Do(req)
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			fmt.Println(err.Error())
+			return
+		}
+	}(res.Body)
+	body, _ := ioutil.ReadAll(res.Body)
+
+	rawResponse := bytes.NewReader(body)
+	decoder := json.NewDecoder(rawResponse)
+	parsedResponse := Response{}
+	err := decoder.Decode(&parsedResponse)
+	if err != nil {
+		fmt.Println(err.Error())
+		return nil
+	}
+
+	for _, r := range parsedResponse.Results {
+		//fmt.Println(r.Ticker)
+		tickers = append(tickers, r.Ticker)
+	}
+	return tickers
+}
+
+func InitialiseStock(ticker string) {
+
+	stocksURL := "https://api.polygon.io/v1/open-close/" + ticker + "/2021-08-04?adjusted=true&apiKey=721mkXq0CBNvCMi5iyJ9E1gBRDiFcT8b"
```

Comment:
Define a configuration object for this to store all the url and api keys data.

Reply:
Pending